UI for custom tokens for Stamen/Stadia maps

paulschuberth opened the pull request:
First of all thank you for this fun application!

This PR goal is resolve #76 by providing a user interface to provide an API key for the maps provided by Stamen/Stadia.

As of right now it's still pretty rough and will likely need some polishing. I am not familiar with the value bindings in `TTPreferencesWindow.xib`. The current solution copies the approach of the other settings and hopes for the best. Maybe some form of validation is appropriate here?

I am very much open to suggestions and ideas. My Obj-C got a bit rusty and I have so far not worked with AppKit.

tomtaylor commented:
Thanks @paulschuberth, this looks like a great start. I think we'd need to do a few things to make this suitable for normal users:

1. Explain what the Stadia token is, and how to get one
2. Prevent selection of the Stadia/Stamen maps until the user has entered one
3. Provide some kind of error if the requests error because of an invalid token

I appreciate that might be biting off more than you want to on this!

paulschuberth commented:
Very valid points! I am happy to give it a try and will update here, once I have something I think is worth reviewing again 👍🏻

paulschuberth commented:
Hi @tomtaylor , thanks again for your review last time around. I believe to have addressed your suggestions. Especially point (2), preventing selection of Stamen maps without a token, proved to be a bit of a challenge.

Looking forward to hear/read what you think about it.

NortonTech3D commented:
Hey Paul, just wondering if this ever got merged into the main branch?

paulschuberth commented:
Ney @NortonTech3D, not to my knowledge, no
